applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -337,6 +374,15 @@ static void handle_dfu_data(const uint8_t *data, size_t size)
 		return;
 	}
 
+	if (IS_ENABLED(CONFIG_DESKTOP_DFU_LOCK)) {
```

Comment:
You could make it `if(IS_ENABLED(CONFIG_DESKTOP_DFU_LOCK) && !dfu_lock_claim() ...{ return}`